### PR TITLE
podman: support to specify the namespace to join using the ns:/PATH notation

### PR DESCRIPTION
**Pull request by giuseppe**
add a prefix `--net ns:/PATH/TO/THE/NAMESPACE` that can be used to specify what namespace must be joined.

We probably want to do the same thing for other namespaces

### CODE REVIEW

**Comment by giuseppe** [comment]
@mheon I might have misunderstood your previous comment about it and how to take rid of it, but I still think we want the extra configuration in `libpod/container.go` so that it can be used when the container is restarted.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by mheon** [comment]
Why do we need to store the information separately when it's in the OCI
spec? All the namespaces and their paths are stored in there, and we'll use
the same base spec every time the container restarts. I don't think we need
to retain the namespace information twice, we just need to make sure that
none of the options that would replace the network namespace with one
podman sets up are set.

**Comment by giuseppe** [comment]
@mheon I've pushed a new version where that information is stored only in the OCI specs.  If this pattern looks good, I am going to add the same feature to other namespaces

**Comment by rhatdan** [comment]
BTW This was a heavily requested feature of Docker that was always rejected. Nice work @giuseppe

File: vendor/github.com/docker/docker/api/types/container/host_config.go [hunk 1]
```diff
@@ -92,15 +92,30 @@ func (n NetworkMode) IsContainer() bool {
 	return len(parts) > 1 && parts[0] == "container"
 }
 
-// ConnectedContainer is the id of the container which network this container is connected to.
-func (n NetworkMode) ConnectedContainer() string {
+// IsNS indicates whether container uses a specific network namespace.
```
Comment (rhatdan) [inline vendor/github.com/docker/docker/api/types/container/host_config.go]: You can't modify the vendored code.  We should probbaly bight the bullet and pull some of this code into a utils. function.

Comment (giuseppe) [reply vendor/github.com/docker/docker/api/types/container/host_config.go]: thanks for noticing it, I completely ignored it was under vendor.  I've pushed a new version with the new functions added to the libpod code

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**